### PR TITLE
build(aio): refactor dgeni packages

### PR DESCRIPTION
**Pull request by petebacondarwin**
This is to tidy up the `author-packagse`, which currently duplicates a
lot of the configuration in the main packages. We need to
DRY this up so that we don't fall foul of a change in one being missed in
the other.

### CODE REVIEW

**Review by wardbell — APPROVED** [review]
Seems like a bit more than a refactoring. But it also looks like a nice step forward and I found nothing evil.

Let's merge it.

**Review by gkalpak — APPROVED** [review]
Just a couple of minor stuff. No blockers. Feel free to ignore altogether.

File: aio/tools/transforms/angular-api-package/index.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+  .processor(require('./processors/markBarredODocsAsPrivate'))
+  .processor(require('./processors/filterPrivateDocs'))
+  .processor(require('./processors/filterIgnoredDocs'))
+
+  // Where do we get the source files?
+  .config(function(readTypeScriptModules, readFilesProcessor, collectExamples) {
+
+    // API files are typescript
+    readTypeScriptModules.basePath = API_SOURCE_PATH;
+    readTypeScriptModules.ignoreExportsMatching = [/^_|ɵ/];
```
Comment (petebacondarwin) [inline aio/tools/transforms/angular-api-package/index.js]: fixed

File: aio/tools/transforms/angular-api-package/index.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  })
+
+  .config(function(convertToJsonProcessor, EXPORT_DOC_TYPES) {
+    const DOCS_TO_CONVERT = EXPORT_DOC_TYPES.concat([
+      'decorator', 'directive', 'pipe', 'module'
+    ]);
+    convertToJsonProcessor.docTypes = convertToJsonProcessor.docTypes.concat(DOCS_TO_CONVERT);
+  });
+
+function requireFolder(folderPath) {
```
Comment (petebacondarwin) [inline aio/tools/transforms/angular-api-package/index.js]: you're right, I missed this one

File: aio/tools/transforms/authors-package/index.js [hunk 3]
```diff
@@ -14,18 +14,20 @@ function createPackage(changedFile) {
     return require('./marketing-package').createPackage();
   }
 
-  const tutorialMatch = /^aio\/content\/tutorial\/|^aio\/content\/examples\/toh-\d/.exec(changedFile);
-  if (tutorialMatch) {
+  const tutorialMatch = /^aio\/content\/tutorial\/([^\.]+)\.md/.exec(changedFile);
```
Comment (petebacondarwin) [inline aio/tools/transforms/authors-package/index.js]: fixed

File: aio/tools/transforms/authors-package/index.js [hunk 4]
```diff
@@ -14,18 +14,20 @@ function createPackage(changedFile) {
...
+  const tutorialExampleMatch = /^aio\/content\/examples\/(toh-[^\/]+)\//.exec(changedFile);
```
Comment (petebacondarwin) [inline aio/tools/transforms/authors-package/index.js]: not bothered :-)

**Pull request merged**